drodrpg/DRODLib/Character.cpp

```diff
@@ -3340,15 +3355,19 @@ void CCharacter::Process(
 
 			case CCharacterCommand::CC_SetPlayerAppearance:
 			{
-				if (this->bIfBlock)
+				if (!bRoomBeingDisplayedOnly)
```

Comment:
My new comment here might have been lost. To restate:
I think we want to perform this check only in the else block, where there might be a side effect, i.e.,
`if (!bRoomBeingDisplayedOnly) pGame->SetPlayerRole(command.x);`

Reply:
Yeah sorry there is some confusion as you have been making comments while I've been working on this so there's pushes happening while you're reviewing. I'm going to go make dinner to give you time to finish reviewing before pushing any more changes.